### PR TITLE
Fix type deprecations

### PR DESCRIPTION
**Pull request by simonster**
Needs JuliaLang/Compat.jl#139

### CODE REVIEW

File: src/JLD.jl [hunk 1]
```diff
@@ -20,6 +20,11 @@ const name_type_attr = "julia type"
 
 typealias BitsKindOrByteString Union(HDF5BitsKind, ByteString)
 
+if VERSION >= v"0.4.0-dev+5379"
+    # Just rename the uses when we drop 0.3 support
+    const UnionType = Union
+end
+
```
Comment (yuyichao) [inline src/JLD.jl]: FYI, I'm trying to add this to Compat.

Comment (yuyichao) [inline src/JLD.jl]: NVM. I was going to add all the missing ones to `Compat` but it seems that this one is not that trival. I'll probably not bother about this one then... ;-p

File: REQUIRE [hunk 2]
```diff
@@ -1,3 +1,3 @@
 julia 0.3.4
 HDF5
-Compat
+Compat 0.7.0+
```
Comment (yuyichao) [inline REQUIRE]: 0.7.1 is tagged.

**Comment by timholy** [comment]
I restarted one of the travis builds, but there's still a problem on 0.3.

File: test/jldtests.jl [hunk 3]
```diff
@@ -243,7 +243,7 @@ function iseq(c1::Base.Sys.CPUinfo, c2::Base.Sys.CPUinfo)
     true
 end
 iseq(x::MyUnicodeStruct☺, y::MyUnicodeStruct☺) = (x.α == y.α && x.∂ₓα == y.∂ₓα)
-iseq(x::Array{None}, y::Array{None}) = size(x) == size(y)
+iseq(x::Array{Union{}}, y::Array{Union{}}) = size(x) == size(y)
```
Comment (yuyichao) [inline test/jldtests.jl]: Need `@compat`

**Comment by yuyichao** [comment]
The test failure happens on master as well https://travis-ci.org/JuliaLang/JLD.jl/jobs/80754899

**Comment by yuyichao** [comment]
A combination of JuliaLang/Compat.jl#143 and  JuliaLang/Compat.jl#144 seems to have fixed the build on 0.3. This should be good to go after updating the Compat version requirement (0.7.3), (and optionally fix the `Union()` deprecation warnins as well).

**Comment by simonster** [comment]
All previous versions of JLD should be changed to require Compat 0.7.3 as well.

**Comment by simonster** [comment]
Also all versions of HDF5 that depend on Compat from before the HDF5/JLD split.

**Comment by yuyichao** [comment]
I think it's probably better to just require `0.7.3` (or even just `0.7.1`) here and then require julia 0.4 in versions `0.7.1` and `0.7.2` of Compat.

**Comment by yuyichao** [comment]
Ref https://github.com/JuliaLang/METADATA.jl/pull/3491 . After it is merged, it should be safe to require any Compat version no lower than 0.7.1 .

**Comment by yuyichao** [comment]
@simonster I merged the commit to hide Compat 0.7.1 and 0.7.2 from Julia 0.3 and updated your commit to require Compat 0.7.3. Will merge when the CI passes again.

**Comment by garrison** [comment]
Is this reason enough for a new release? :)

**Comment by yuyichao** [comment]
There's still #27

**Pull request merged**